src/components/v5/common/CompletedAction/partials/PaymentBuilder/partials/PaymentBuilderWidget/utils.ts

```diff
@@ -9,7 +12,47 @@ export const getExpenditureStep = (status?: ExpenditureStatus) => {
       return ExpenditureStep.Review;
     case ExpenditureStatus.Locked:
       return ExpenditureStep.Funding;
+    case ExpenditureStatus.Finalized:
+      return ExpenditureStep.Payment;
     default:
       return ExpenditureStep.Create;
   }
 };
+
+/**
+ * Returns a boolean indicating whether the expenditure is fully funded,
+ * i.e. the balance of each token is greater than or equal to the sum of its payouts
+ */
+export const isExpenditureFullyFunded = (expenditure?: Expenditure | null) => {
+  if (!expenditure) {
+    return false;
+  }
+
+  if (!expenditure.balances) {
+    return false;
+  }
+
+  const amountsByToken = expenditure.slots.reduce<Record<string, BigNumber>>(
+    (amounts, slot) => {
+      slot.payouts?.forEach((payout) => {
+        if (!payout.isClaimed) {
+          // eslint-disable-next-line no-param-reassign
+          amounts[payout.tokenAddress] = BigNumber.from(
+            amounts[payout.tokenAddress] ?? 0,
+          ).add(payout.amount);
+        }
+      });
+
+      return amounts;
+    },
+    {},
+  );
```

Comment:
This is improper use of `reduce`. There no case where you actually modify or increment the accumulator, you always return the the same object.

This is at best a use case for `map` rather.